pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -541,6 +576,40 @@ func removeFinalizerFromCRDInstance(ctx context.Context,
 	}
 }
 
+// addFinalizerToPVC will add the CNS Finalizer, cns.vmware.com/pvc-protection,
+// from a given PersistentVolumeClaim.
+func addFinalizerToPVC(ctx context.Context, client client.Client, pvc *v1.PersistentVolumeClaim) error {
+	log := logger.GetLogger(ctx)
+	pvc.Finalizers = append(pvc.Finalizers, cnsoperatortypes.CNSPvcFinalizer)
+	log.Debugf("Adding %q finalizer on PersistentVolumeClaim: %q on namespace: %q",
+		cnsoperatortypes.CNSPvcFinalizer, pvc.Name, pvc.Namespace)
+	err := client.Update(ctx, pvc)
+	if err != nil {
```

Comment:
Do you have to handle a case where SV PVC is not present?

Reply:
Yes, we check if getPVC() succeeds or fails early in the reconciliation.